### PR TITLE
Core: add `UnsafeMutableRawPointer`

### PR DESCRIPTION
**Pull request by MaxDesiatov**

### CODE REVIEW

File: Sources/Core/UnsafeMutableRawPointer.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  @_transparent
+  public init?(@_nonEphemeral mutating other: UnsafeRawPointer?) {
+    guard let unwrapped = other else { return nil }
+    _rawValue = unwrapped._rawValue
+  }
+
+  @inlinable
+  public static func allocate(byteCount: Int,
+                              alignment: Int) -> UnsafeMutableRawPointer {
```
Comment (compnerd) [inline Sources/Core/UnsafeMutableRawPointer.swift:40]: An alternate wrapping which would also work:

```suggestion
  public static func allocate(byteCount: Int, alignment: Int)
      -> UnsafeMutableRawPointer {
```

File: Sources/Core/UnsafeMutableRawPointer.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    var alignment = alignment
+    if alignment <= minAllocationAlignment() {
+      alignment = 0
+    }
```
Comment (compnerd) [inline Sources/Core/UnsafeMutableRawPointer.swift:44]: I would just write this as:

```suggestion
    let alignment: Int =
        alignment <= minAllocationAlignment() ? 0 : alignment
```

(or on one line if it fits).

File: Sources/Core/UnsafeMutableRawPointer.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  @discardableResult
+  public func bindMemory<T>(to type: T.Type, capacity count: Int)
+    -> UnsafeMutablePointer<T> {
+    Builtin.bindMemory(_rawValue, count._value, type)
+    return UnsafeMutablePointer<T>(_rawValue)
+  }
+
+  @_transparent
+  public func assumingMemoryBound<T>(to _: T.Type) -> UnsafeMutablePointer<T> {
+    UnsafeMutablePointer<T>(_rawValue)
```
Comment (compnerd) [inline Sources/Core/UnsafeMutableRawPointer.swift:66]:
```suggestion
    return UnsafeMutablePointer<T>(_rawValue)
```

Comment (MaxDesiatov) [reply Sources/Core/UnsafeMutableRawPointer.swift]: LMK if you'd like me to fixup your suggestions commit into parent.

File: Sources/Core/UnsafeMutableRawPointer.swift [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  public func deallocate() {
+    let size: Int = -1
+    let alignment: Int = 0
+    Builtin.deallocRaw(_rawValue, size._value, alignment._value)
+  }
+
+  @_transparent
+  @discardableResult
+  public func bindMemory<T>(to type: T.Type, capacity count: Int)
+    -> UnsafeMutablePointer<T> {
```
Comment (compnerd) [inline Sources/Core/UnsafeMutableRawPointer.swift:57]:
```suggestion
      -> UnsafeMutablePointer<T> {
```

**Review by compnerd — APPROVED** [review]
It's fine, we can squash it on merge.

**Pull request merged**